homeassistant/components/climate/__init__.py

```diff
@@ -717,6 +730,34 @@ def async_turn_aux_heat_off(self):
         """
         return self.hass.async_add_job(self.turn_aux_heat_off)
 
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        # For now, we convert from the legacy technique to the new
+        # supported_features. This follows exactly the way of determining
+        # whether an attribute is included in state_attributes()
+
+        support = SUPPORT_TARGET_TEMPERATURE
+        if self.target_temperature_high is not None:
+            support |= SUPPORT_TARGET_TEMPERATURE_HIGH
+            support |= SUPPORT_TARGET_TEMPERATURE_LOW
+        if self.target_humidity:
+            support |= SUPPORT_TARGET_HUMIDITY
+        if self.current_fan_mode is not None:
+            support |= SUPPORT_FAN_MODE
+        if self.current_operation is not None:
+            support |= SUPPORT_OPERATION_MODE
+        if self.current_hold_mode is not None:
+            support |= SUPPORT_HOLD_MODE
+        if self.current_swing_mode is not None:
+            support |= SUPPORT_SWING_MODE
+        if self.is_away_mode_on is not None:
+            support |= SUPPORT_AWAY_MODE
+        if self.is_aux_heat_on is not None:
+            support |= SUPPORT_AUX_HEAT
+
+        return support
```

Comment:
Things like this will be never removed later. I think we need update the other platforms to return the correct supported features.

Reply:
That would mean changing *all* climate platforms in this PR. Frankly, I wouldn't bet on managing that without breaking some of them, especially since I can't test most of them. Do you still think I should do it? It's a fairly mechanical translation of what properties the platforms currently set into the flags, shouldn't be too much work.

Reply:
I prefer to update all of them at once, move to the new way of doing things and enforce it.